File: day28/mountain_array_checker.py
def is_mountain_array(arr):
    print("Input array:", arr)
    
    n = len(arr)
    if n < 3:
        return False

    i = 0

    # Ascend
    while i + 1 < n and arr[i] < arr[i + 1]:
        i += 1

    # Peak can't be first or last
    if i == 0 or i == n - 1:
        return False

    # Descend
    while i + 1 < n and arr[i] > arr[i + 1]:
        i += 1

    return i == n - 1

# Example usage
examples = [
    [0, 3, 2, 1],
    [2, 1],
    [3, 5, 5],
    [0, 1, 2, 3, 2, 1, 0]
]

for ex in examples:
    print("Is mountain:", is_mountain_array(ex))
    print("---")

File: day28/mountain_array_checker.rs
fn is_mountain_array(arr: &[i32]) -> bool {
    println!("Input array: {:?}", arr);
    let n = arr.len();
    if n < 3 {
        return false;
    }

    let mut i = 0;

    // Ascend
    while i + 1 < n && arr[i] < arr[i + 1] {
        i += 1;
    }

    // Peak can't be first or last
    if i == 0 || i == n - 1 {
        return false;
    }

    // Descend
    while i + 1 < n && arr[i] > arr[i + 1] {
        i += 1;
    }

    i == n - 1
}

fn main() {
    let test_cases = vec![
        vec![0, 3, 2, 1],
        vec![2, 1],
        vec![3, 5, 5],
        vec![0, 1, 2, 3, 2, 1, 0],
    ];

    for case in test_cases {
        let result = is_mountain_array(&case);
        println!("Is mountain: {}\n---", result);
    }
}
